From 82f405871edb530a0a93cc2a5aefc698087eccc2 Mon Sep 17 00:00:00 2001
From: Lu Yahan <yahan@iscas.ac.cn>
Date: Fri, 28 May 2021 17:29:47 +0800
Subject: [PATCH] [riscv64][codegen] Assert that RegisterArray has only unique
 registers

Port 032ab3d2ecca1dd5286295164f5f5e603a8ee774

Change-Id: I55715e1943e0e178079e810b8979d14d84e776cf
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/2924605
Auto-Submit: Yahan Lu <yahan@iscas.ac.cn>
Commit-Queue: Brice Dobry <brice.dobry@futurewei.com>
Reviewed-by: Brice Dobry <brice.dobry@futurewei.com>
Cr-Commit-Position: refs/heads/master@{#74843}
---
 src/codegen/riscv64/interface-descriptors-riscv64-inl.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/codegen/riscv64/interface-descriptors-riscv64-inl.h b/src/codegen/riscv64/interface-descriptors-riscv64-inl.h
index ce9e41be85..b8a61f774c 100644
--- a/src/codegen/riscv64/interface-descriptors-riscv64-inl.h
+++ b/src/codegen/riscv64/interface-descriptors-riscv64-inl.h
@@ -22,11 +22,13 @@ constexpr auto CallInterfaceDescriptor::DefaultRegisterArray() {
 
 // static
 constexpr auto WriteBarrierDescriptor::registers() {
-  return RegisterArray(a0, a1, a2, a3, kReturnRegister0);
+  STATIC_ASSERT(kReturnRegister0 == a0);
+  return RegisterArray(a0, a1, a2, a3);
 }
 
 // static
 constexpr auto DynamicCheckMapsDescriptor::registers() {
+  STATIC_ASSERT(kReturnRegister0 == a0);
   return RegisterArray(kReturnRegister0, a1, a2, a3, cp);
 }
 
-- 
2.35.1

